neqo-http3/src/connection_client.rs

```diff
@@ -30,21 +32,30 @@ use std::time::Instant;
 
 use crate::{Error, Res};
 
-// This is used for filtering send_streams and recv_Streams with a stream_ids greater that a given id.
-fn id_gte<T, U>(base: T) -> impl FnMut((&T, &U)) -> Option<T> + 'static
+// This is used for filtering send_streams and recv_Streams with a stream_ids greater than or equal a given id.
+// Only the same type (bidirectional or unidirectionsl) streams are filtered.
+fn id_gte<U>(base: StreamId) -> impl FnMut((&u64, &U)) -> Option<u64> + 'static
```

Comment:
the whole http3 code uses u64 instead of StreamId for the beginning(the reason was that I did not want to expose StreamId to applications using http3, and it was better, at the time to use u64 in whole http3 code....). We can change that, but let's do it in a different PR.